core/market/decentralized/src/db/models/proposal.rs

```diff
@@ -141,6 +122,56 @@ impl Proposal {
         }
     }
 
+    pub fn new_provider_initial(demand: ModelDemand, offer: ModelOffer) -> Proposal {
```

Comment:
Ok, but could you suggest different name, that tells that it is the first proposal that Provider encountered?
Because we will have second function here, that will deal with NOT first proposals, and I reserve this name for them ;)

Reply:
`new` ;)

Reply:
I will try to do something with it in https://github.com/golemfactory/yagna/pull/412